Sources/AudioKit/Taps/FFTTap.swift

```diff
@@ -35,75 +44,126 @@ open class FFTTap: BaseTap {
     override open func doHandleTapBlock(buffer: AVAudioPCMBuffer, at time: AVAudioTime) {
         guard buffer.floatChannelData != nil else { return }
 
-        fftData = FFTTap.performFFT(buffer: buffer, isNormalized: isNormalized, zeroPaddingFactor: zeroPaddingFactor)
+        fftData = FFTTap.performFFT(buffer: buffer,
+                                    isNormalized: isNormalized,
+                                    zeroPaddingFactor: zeroPaddingFactor,
+                                    fftSetupForBinCount: fftSetupForBinCount)
         handler(fftData)
     }
 
-    static func performFFT(buffer: AVAudioPCMBuffer, isNormalized: Bool = true, zeroPaddingFactor: UInt32 = 0) -> [Float] {
+    static func performFFT(buffer: AVAudioPCMBuffer,
+                           isNormalized: Bool = true,
+                           zeroPaddingFactor: UInt32 = 0,
+                           fftSetupForBinCount: FFTSetupForBinCount? = nil) -> [Float] {
         let frameCount = buffer.frameLength + buffer.frameLength * zeroPaddingFactor
-        let log2n = UInt(round(log2(Double(frameCount))))
-        let bufferSizePOT = Int(1 << log2n)
-        let inputCount = bufferSizePOT / 2
+        let log2n = determineLog2n(frameCount: frameCount, fftSetupForBinCount: fftSetupForBinCount)
+        let bufferSizePOT = Int(1 << log2n) // 1 << n = 2^n
+        let inputCount = bufferSizePOT / 2 // number of bins returned
+
         let fftSetup = vDSP_create_fftsetup(log2n, Int32(kFFTRadix2))
```

Comment:
for a future optimization, apparently it's expensive to run this each time. fftSetup can be turned into a property on this class and reused in each call (since log2n won't change between each run)

Reply:
100% agree! I actually had a comment about that on this line, but removed it before putting in the PR. We should do that